### PR TITLE
Replace Photoswipe react library

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-frontend/issues/9659
Fixes https://github.com/mozilla/addons-frontend/issues/10145

---

It should be _very_ similar to what we had before.

### CODE REVIEW

File: src/amo/components/ScreenShots/index.js [hunk 1]
```diff
@@ -1,26 +1,13 @@
 /* @flow */
-/* global window */
-import invariant from 'invariant';
 import * as React from 'react';
-import { PhotoSwipeGallery } from 'react-photoswipe';
-import 'react-photoswipe/lib/photoswipe.css';
+import { Gallery, Item } from 'react-photoswipe-gallery';
+import invariant from 'invariant';
```
Comment (bobsilverberg) [inline src/amo/components/ScreenShots/index.js:4]: Minor nit: I try to keep these in alphabetical order.

File: src/amo/components/ScreenShots/index.js [hunk 2]
```diff
@@ -30,67 +17,32 @@ export const PHOTO_SWIPE_OPTIONS = {
   counterEl: true,
   arrowEl: true,
   preloaderEl: true,
-  // Overload getThumbBoundsFn as workaround to
```
Comment (bobsilverberg) [inline src/amo/components/ScreenShots/index.js:33]: Nice we could get rid of this!

File: src/amo/components/ScreenShots/index.js [hunk 3]
```diff
@@ -103,13 +55,36 @@ export default class ScreenShots extends React.Component<Props> {
...
+                  original={preview.src}
+                  thumbnail={preview.thumbnail_src}
+                  width={preview.w}
+                  height={preview.h}
+                  title={preview.title}
+                >
+                  {({ ref, open }) => (
+                    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
+                    <img
+                      alt=""
```
Comment (bobsilverberg) [inline src/amo/components/ScreenShots/index.js:75]: I notice that in the old code, the `img` has an `alt` attribute of `preview.title`. Do we not want that anymore? Is that dealt with via the `title` property of the `Item` above?

File: src/amo/components/ScreenShots/styles.scss [hunk 4]
```diff
@@ -7,10 +7,6 @@ $image-height: $screenshot-height - 20px;
 $screenshot-width: 320px;
 
 .ScreenShots {
-  .pswp-thumbnails {
-    line-height: 0;
-  }
-
   .pswp__caption__center {
```
Comment (bobsilverberg) [inline src/amo/components/ScreenShots/styles.scss:10]: I see you removed the references to the `.pswp-thumbnails` class. I take it that the other classes that start with `.pswp` are still relevant?

**Review by bobsilverberg — APPROVED** [review]
Looks good, thanks for taking care of this @willdurand.

r+wc

File: src/amo/components/ScreenShots/index.js (hunk 3 above)
Comment (willdurand) [reply src/amo/components/ScreenShots/index.js:75]: I think setting `alt` to `preview.title` is a good idea. `title` was used for the caption in fullscrene but maybe `alt` could be used as well? Or we could have both. I'll update the PR, thanks!

File: src/amo/components/ScreenShots/styles.scss (hunk 4 above)
Comment (willdurand) [reply src/amo/components/ScreenShots/styles.scss:10]: Yes, `pswp-thumbnails` was part of the old _react_ library, the others seem to be from PhotoSwipe itself. At least, I could find them in the generated HTML.

File: src/amo/components/ScreenShots/index.js (hunk 1 above)
Comment (willdurand) [reply src/amo/components/ScreenShots/index.js:4]: thanks, I like that idea a lot!

**Pull request merged**